xpk.py

```diff
@@ -3408,8 +3506,97 @@ def run_gke_node_pool_create_command(
         f'{args.cluster}-np-{slice_num}' for slice_num in range(args.num_slices)
     ]
 
+  node_pools_to_remain = []
+  delete_commands = []
+  delete_task_names = []
+  if existing_node_pool_names:
+    return_code, existing_node_pool_zone = get_nodepool_zone(
+        args, existing_node_pool_names[0]
+    )
+    if return_code != 0:
+      xpk_exit(1)
+
+    if existing_node_pool_zone and existing_node_pool_zone != args.zone:
+      xpk_print(
+          f'Cluster {args.cluster} already has nodepools in zone:'
+          f' {existing_node_pool_zone}. Use the same zone to update nodepools'
+          ' in the cluster.'
+      )
+      xpk_exit(1)
+
+    node_pools_to_delete = get_node_pools_to_delete(
+        args, existing_node_pool_names, desired_node_pool_names
+    )
+    for node_pool_name in existing_node_pool_names:
+      if node_pool_name.find(f'{args.cluster}-np-') != 0:
+        continue
+
+      if node_pool_name in node_pools_to_delete:
+        command = (
+            'gcloud beta container node-pools delete'
+            f' {node_pool_name} --cluster={args.cluster}'
+            f' --zone={zone_to_region(args.zone)}'
+            f' --project={args.project} --quiet'
+        )
+        task = f'NodepoolDelete-{node_pool_name}'
+        delete_commands.append(command)
+        delete_task_names.append(task)
+      else:
+        node_pools_to_remain.append(node_pool_name)
+
+  # Deletion of nodepools should happen before attempting to create new nodepools for the case
+  # when cluster is getting updated from 'x' device_type to 'y' device_type. In that case,
+  # '{args.cluster}-np-i' nodepool will be re-created for 'y' device_type.
+  if delete_commands:
+    will_delete = True
+    if node_pools_to_delete and not args.force:
+      will_delete = get_user_input(
+          f'Planning to delete {len(node_pools_to_delete)} node pools including'
+          f' {node_pools_to_delete}. \nDo you wish to delete: y (yes) / n'
+          ' (no):\n'
+      )
+    if not will_delete:
+      xpk_print(
+          'You have requested to not delete the existing nodepools in the'
+          ' cluster. There will be no change to the cluster.'
+      )
+      xpk_exit(1)
```

Comment:
change to return 1 to allow parent function to exit to follow our current exit pattern. I saw a few other cases that can be changed as well.

Reply:
Updated!

Reply:
Done!